web/html/src/manager/admin/config/monitoring.js

```diff
@@ -0,0 +1,324 @@
+      keys.map(key => 
+        (<li key={key}>
+          <ExporterIcon status={props.exportersStatus[key]}/>
+          { key in exporterMap ? exporterMap[key] : capitalize(key)}
+          </li>)
+        )
+    }
+    </ul>;
+}
+
+type MonitoringAdminProps = {
+  loading: boolean,
+  monitoringEnabled: ?boolean,
+  exportersStatus: ?{[string]: boolean},
+  messages: Array<Object>,
+  onChange: () => void
+}
+
+class MonitoringAdmin extends React.Component<MonitoringAdminProps> {
+  
+  render() {
+      let buttonLoadingText : string = "";
+      if (this.props.loading) {
+        if (this.props.monitoringEnabled === null) {
+          buttonLoadingText = t("Checking monitoring services...");
+        } else if (this.props.monitoringEnabled === true) {
+          buttonLoadingText = t("Disabling monitoring services...");
+        } else if (this.props.monitoringEnabled === false) {
+          buttonLoadingText = t("Enabling monitoring services...");
+        }
+      }
+      return (
+      <div className="responsive-wizard">
+        <Messages items={this.props.messages}/>
+        <div className="spacewalk-toolbar-h1">
+          <div className="spacewalk-toolbar"></div>
+          <h1>
+            <i className="fa fa-info-circle"></i>
+            {t("SUSE Manager Configuration - Monitoring")}
+            <a href="/rhn/help/reference/en-US/ref.webui.admin.config.jsp#s3-sattools-config-gen" target="_blank">
+              <i className="fa fa-question-circle spacewalk-help-link"></i>
+            </a>
+          </h1>
+        </div>
+        <div className="page-summary">
+          <p>
+            {t("Setup your SUSE Manager server monitoring.")}
+          </p>
+        </div>
+        <div className='spacewalk-content-nav'>
+          <ul className="nav nav-tabs">
+            <li><a href="/rhn/admin/config/GeneralConfig.do?">General</a></li>
+            <li><a href="/rhn/admin/config/BootstrapConfig.do?">Bootstrap Script</a></li>
+            <li><a href="/rhn/admin/config/Orgs.do?">Organizations</a></li>
+            <li><a href="/rhn/admin/config/Restart.do?">Restart</a></li>
+            <li><a href="/rhn/admin/config/Cobbler.do?">Cobbler</a></li>
+            <li><a href="/rhn/admin/config/BootstrapSystems.do?">Bare-metal systems</a></li>
+            <li className="active"><a href="/rhn/manager/admin/config/monitoring?">Monitoring</a></li>
+          </ul>
+        </div>
+        <Panel
+          key="schedule"
+          title={t('Monitoring')}
+          headingLevel="h4"
+          footer={
+            <div className="row">
+              <div className="col-md-offset-3 col-md-9">
+                { this.props.loading ? 
+                  <Button id="loading-btn" disabled={true} 
+                    className="btn-default" icon="fa-circle-o-notch fa-spin" text={buttonLoadingText}/> :
+                  <AsyncButton id="monitoring-btn" defaultType="btn-success"
```

Comment:
While I like the idea of toggling the button behavior based on the current status, I think we don't have this concept spread around the product. At the same time this is a *special* slice of the product, so I think we can keep it in the end.

I would suggest the following 2 options anyway:
1- split the button into two, one to enable and one to disable (there is a lot of space there I think it's not a big deal)
2- keep the single button but change the color depending on the status. In this case I suggest `green` for `Enable...` and `gray` for `Disable.....` (`gray` and not `red` because it just disable it, it does not delete anything IIUC).

And 2 other small tips to the button:
1- maybe the `monitoring` word can be omitted, like `Disable services` only: we are in the `Monitoring` tab in the end
2- if `enable` and `disable` can be defined as a `start` and `stop` actions, why not adding a small icon at the left of the text? I'd say the `►` icon and the `■` icon

What do you think?